src/util/CompositeAsyncHandler.ts

```diff
@@ -0,0 +1,29 @@
+import { AsyncHandler } from './AsyncHandler';
+
+export class CompositeAsyncHandler<TIn, TOut> implements AsyncHandler<TIn, TOut> {
+  private readonly handlers: AsyncHandler<TIn, TOut>[];
+
+  public constructor (handlers: AsyncHandler<TIn, TOut>[]) {
+    this.handlers = handlers;
+  }
+
+  public async canHandle (input: TIn): Promise<boolean> {
```

Comment:
Would not re-call in `handle`.

Reply:
@RubenVerborgh But for a composite handler it needs to be called again `handle` since it doesn't know which one passed the test in `canHandle`. I thought about adding a "cache" that remembers the last handler that returned true in the last `canHandle` call, but there could be sync errors there if the composite handler is used by multiple classes. Could do more advanced caching based on the input but I think then it's becoming to convoluted for what it is.

The idea would then indeed have to be that the `canHandle` checks are always cheap. But I think that was the goal anyway, to not have handlers as specific as in Comunica.

Reply:
> But for a composite handler it needs to be called again `handle`

Good point indeed; a cache _might_ be used if needed, but I indeed expect cheap checks.
Also, the _generic_ composite handler might be naive, in that it just linearly asks every handler. But we could have a more specific handler (or inject a selection algorithm) that can decide based on specific `TIn`; for instance, a `switch` on HTTP method.